sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -342,6 +359,12 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
             "Hive metastore in Spark SQL specific format, which is NOT compatible with Hive. "
         (None, message)
 
+      case _ if currentSessionConf(SQLConf.CASE_SENSITIVE) =>
```

Comment:
Ok, I'll remove this change. The write-path change you propose isn't necessary because if you have an "invalid" schema (same column name with different case), the Hive metastore will complain and the table will be stored as non-Hive-compatible.

The problem this was trying to avoid is related to the changes in `alterTableSchema`; if you create a Hive-compatible table here, then later tried to update it with an invalid schema, you'd have a frankentable because the code in `alterTableSchema` was wrong.

But since this change is mainly about fixing `alterTableSchema`, you'll now get a proper error in that case instead of ending up with a potentially corrupted table.